pkg/controller/daemon/daemon_controller.go

```diff
@@ -1200,6 +1203,27 @@ func (dsc *DaemonSetsController) updateDaemonSetStatus(ds *apps.DaemonSet, nodeL
 		} else {
 			if scheduled {
 				numberMisscheduled++
+				desiredNumberScheduled++
```

Comment:
This section doesn't seem right. If pod is scheduled onto a node where it should not be, I think we should only count that one in the `numberMisscheduled` variable. In other words, if the pod is misscheduled, we no longer cares if the pod is ready or available.